ctlearn/run_model.py

```diff
@@ -506,10 +358,66 @@ def write(prediction):
 
     args = parser.parse_args()
 
-    with open(args.config_file, 'r') as config_file:
-        config = yaml.safe_load(config_file)
-
-    run_model(config, mode=args.mode, debug=args.debug, log_to_file=args.log_to_file)
+    random_seeds = []
+    for run in np.arange(args.multiple_runs):
+        with open(args.config_file, 'r') as config_file:
+            config = yaml.safe_load(config_file)
+
+        # Overwrite the random seed in the config file
+        if args.multiple_runs == 1 and args.random_seed != 0:
+            if 1000 <= args.random_seed <= 9999:
+                config['Data']['seed'] = args.random_seed
+                config['Logging']['add_seed'] = True
+            else:
+                raise ValueError("Random seed: '{}'. "
+                             "Must be 4 digit integer!".format(
+                                 args.random_seed))
+        random_seed = config['Data']['seed']
+
+        # Create and overwrite the random seed in the config file for multiple runs
+        if args.multiple_runs != 1:
+            while True:
+                random_seed = randint(1000,9999)
+                if random_seed not in random_seeds:
+                    random_seeds.append(random_seed)
+                    break
+            config['Data']['seed'] = random_seed
+            config['Logging']['add_seed'] = True
+            print("CTLearn run {} with random seed '{}':".format(run+1,config['Data']['seed']))
+
+        if args.mode == 'train' or args.mode == 'load_only':
+            run_model(config, mode=args.mode, debug=args.debug, log_to_file=args.log_to_file)
+        elif args.mode == 'predict':
+            for key in config['Prediction']['prediction_file_lists']:
+                with open(args.config_file, 'r') as config_file:
+                    config = yaml.safe_load(config_file)
+                config['Data']['seed'] = random_seed
+                if args.multiple_runs != 1 or args.random_seed != 0:
+                    config['Logging']['add_seed'] = True
+                config['Data']['shuffle'] = False
+                config['Prediction']['prediction_label'] = key
+                run_model(config, mode=args.mode, debug=args.debug, log_to_file=args.log_to_file)
+        elif args.mode == 'train_and_predict':
+            run_model(config, mode='train', debug=args.debug, log_to_file=args.log_to_file)
+            for key in config['Prediction']['prediction_file_lists']:
+                with open(args.config_file, 'r') as config_file:
+                    config = yaml.safe_load(config_file)
+                config['Data']['seed'] = random_seed
+                if args.multiple_runs != 1 or args.random_seed != 0:
+                    config['Logging']['add_seed'] = True
+                config['Data']['shuffle'] = False
+                config['Prediction']['prediction_label'] = key
+                run_model(config, mode='predict', debug=args.debug, log_to_file=args.log_to_file)
```

Comment:
```suggestion
        if args.mode in ['train', 'train_and_predict', 'load_only']:
            run_model(config, mode=args.mode, debug=args.debug, log_to_file=args.log_to_file)
        if args.mode in ['predict', 'train_and_predict']:
            for key in config['Prediction']['prediction_file_lists']:
                with open(args.config_file, 'r') as config_file:
                    config = yaml.safe_load(config_file)
                config['Data']['seed'] = random_seed
                if args.multiple_runs != 1 or args.random_seed != 0:
                    config['Logging']['add_seed'] = True
                config['Data']['shuffle'] = False
                config['Prediction']['prediction_label'] = key
                run_model(config, mode=args.mode, debug=args.debug, log_to_file=args.log_to_file)
```

Reply:
Thanks for the suggestion! I sightly modified it to make it work. The run_model function not have a 'train_and_predict' mode, so we have to call it recursively and manually.

I search for "train" and "predict" in the args.mode string. So it will also work accordingly if the user set the mode to "training", "predicting", "prediction", "train_and_predict", "training_and_predicting" ...
The mode "load_only" is separated.

What do you think?

Reply:
Clever! Seems good to me.